packages/plugin-types/src/amplify_resource_group_name.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Represents the types of resource group name
+ */
+export type AmplifyResourceGroupName =
+  | 'auth'
+  | 'conversationHandlerFunction'
+  | 'data'
+  | 'function'
```

Comment:
same with function i think only core resource groups should be here, auth data and storage

Reply:
I was thinking to add all the resource groups we define to be consistent across the board. Also in the case where customers change the resource group of a function and want to change it back the resource group for functions is discoverable.

Reply:
I think narrowing down this makes sense given that arbitrary string can also be provided. This can be expanded later.

The consistency of our internal stuff is secondary concern here.